go/vendor/github.com/keybase/stellarnet/format.go

```diff
@@ -54,19 +54,21 @@ func FmtAmount(amount string, precisionTwo bool, rounding FmtRoundingBehavior) (
 	if amount == "" {
 		return "", fmt.Errorf("empty amount")
 	}
-	x, err := ParseAmount(amount)
+	x, err := parseAmountIntoDecimal(amount)
 	if err != nil {
 		return "", fmt.Errorf("unable to parse amount %s: %v", amount, err)
 	}
-	precision := 7
+	var precision int32 = 7
 	if precisionTwo {
 		precision = 2
 	}
 	var s string
 	if rounding == Round {
-		s = x.FloatString(precision)
+		s = x.StringFixed(precision)
 	} else {
-		s = x.FloatString(precision + 1)
+		// bigdecimal is used instead of big.Rat because big.Rat.FloatString rounds the last digit.
+		// Which would turn 2.9999999 into 3.0 when we want 2.9.
```

Comment:
is this an extreme example, or is it always this aggressive in chopping off pennies?

Reply:
Both the example you deleted and the one you replaced it with are accurate, eh?
https://goplay.space/#qqTM1nZco_z

Reply:
The issue with FloatString was revealed to me in two stages.
First, the second least significant digit out of FloatString can change due to rounding: `(2.1299).FloatString(3)[:-1] -> 2.13`.
Second, all the digits out of FloatString can change due to rounding: `(2.9999).FloatString(3)[:-1] -> 3.00`

Reply:
huh. ok.